### ISSUE TITLE
Yandex.Music doesn't work after upgrade to from 3.0.2 -> 3.0.4

### ISSUE THREAD

**Issue by petRUShka** [tiliado/nuvola-app-yandex-music#3]
Console log after click Yandex.Music button (and I see blank screen after it):

```
Runner: [WARNING  libnotify] Failed to connect to proxy
Worker: [DEBUG    Nuvola] WebExtension.vala:220: Page 1 created for (null)
Worker: [DEBUG    Nuvola] WebExtension.vala:291: Document loaded about:loading
Runner: Error loading module '/usr/lib/mozilla/plugins/nppdf.so': /usr/lib/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: Error loading module '/usr/lib64/mozilla/plugins/nppdf.so': /usr/lib64/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib64/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: [WARNING  Nuvola] FormatSupportCheck.vala:158: Too many Flash plugins have been found: 2
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

Runner: CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-enI9IPrAhjQ8o77+oTojVQ==''. It will be ignored.
Runner: https://music.yandex.ru/search?text=the%20xx:34: CONSOLE ERROR [Report Only] Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval' vk.com cdn.pushwoosh.com yandex.ru yandex.st yandex.net yastatic.net yandexadexchange.net *.yandex.ru *.yandex.net *.yastatic.net *.yandexadexchange.net *.yandex-team.ru 'nonce-enI9IPrAhjQ8o77+oTojVQ=='".

Worker: [DEBUG    Nuvola] WebExtension.vala:130: Window object cleared for 'https://music.yandex.ru/search?text=the%20xx'
Worker: [DEBUG    Nuvola] WebExtension.vala:140: Init frame: https://music.yandex.ru/search?text=the%20xx, 0x18c1e70, 0x16eae20, 0x7f92edb5b140
Runner: https://music.yandex.ru/api/v2.1/index/music.yandex.ru:1:2719: CONSOLE ERROR TypeError: undefined is not an object (evaluating 'AudioContext.prototype')
Runner: [CRITICAL  Json] json_node_get_parent: assertion 'JSON_NODE_IS_VALID (node)' failed
Runner: [CRITICAL  Json] Application will not function properly.
Runner: [CRITICAL  GLib] g_ptr_array_remove_index: assertion 'index_ < rarray->len' failed
Runner: [CRITICAL  GLib] Application will not function properly.
Runner: https://music.yandex.ru/index.ru.js?v=0.14.02rc01:24:3602: CONSOLE ERROR ReferenceError: Can't find variable: ya
Runner: https://music.yandex.ru/search?text=the%20xx:46: CONSOLE ERROR [Report Only] Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval' vk.com cdn.pushwoosh.com yandex.ru yandex.st yandex.net yastatic.net yandexadexchange.net *.yandex.ru *.yandex.net *.yastatic.net *.yandexadexchange.net *.yandex-team.ru 'nonce-enI9IPrAhjQ8o77+oTojVQ=='".

Runner: https://music.yandex.ru/search?text=the%20xx:47:20: CONSOLE ERROR TypeError: Mu.init is not a function. (In 'Mu.init()', 'Mu.init' is undefined)
Worker: [DEBUG    Nuvola] WebExtension.vala:291: Document loaded https://music.yandex.ru/search?text=the%20xx
Runner: Error loading module '/usr/lib/mozilla/plugins/nppdf.so': /usr/lib/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: Error loading module '/usr/lib64/mozilla/plugins/nppdf.so': /usr/lib64/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib64/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: [CRITICAL  Json] json_node_get_parent: assertion 'JSON_NODE_IS_VALID (node)' failed
Runner: [CRITICAL  Json] Application will not function properly.
Runner: [CRITICAL  GLib] g_ptr_array_remove_index: assertion 'index_ < rarray->len' failed
Runner: [CRITICAL  GLib] Application will not function properly.
```

P.S. This is the same with Yandex.Music 1.1 and Yandex.Music 1.2.

**Comment by petRUShka** [comment]
Compare with 3.0.2 nuvola logs (working version):

```
Runner: [WARNING  GLib-GObject] g_object_set_property: construct property "local-storage-directory" for object 'WebKitWebContext' can't be set after construction
Runner: [WARNING  libnotify] Failed to connect to proxy
Worker: [DEBUG    Nuvola] WebExtension.vala:242: Page 1 created for (null)
Worker: [DEBUG    Nuvola] WebExtension.vala:192: Waiting until initialized
Runner: Error loading module '/usr/lib/mozilla/plugins/nppdf.so': /usr/lib/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: Error loading module '/usr/lib64/mozilla/plugins/nppdf.so': /usr/lib64/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib64/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: [WARNING  Nuvola] FormatSupportCheck.vala:158: Too many Flash plugins have been found: 2
Runner: [CRITICAL  Json] json_node_get_parent: assertion 'JSON_NODE_IS_VALID (node)' failed
Runner: [CRITICAL  Json] Application will not function properly.
Runner: [CRITICAL  GLib] g_ptr_array_remove_index: assertion 'index_ < rarray->len' failed
Runner: [CRITICAL  GLib] Application will not function properly.
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

Runner: CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-iVHOUO9uBbRPCtlpTC5s7Q==''. It will be ignored.
Runner: [CRITICAL  Json] json_node_get_parent: assertion 'JSON_NODE_IS_VALID (node)' failed
Runner: [CRITICAL  Json] Application will not function properly.
Runner: [CRITICAL  GLib] g_ptr_array_remove_index: assertion 'index_ < rarray->len' failed
Runner: [CRITICAL  GLib] Application will not function properly.
Runner: https://music.yandex.ru/search?text=the%20xx:34: CONSOLE ERROR [Report Only] Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval' vk.com cdn.pushwoosh.com yandex.ru yandex.st yandex.net yastatic.net yandexadexchange.net *.yandex.ru *.yandex.net *.yastatic.net *.yandexadexchange.net *.yandex-team.ru 'nonce-iVHOUO9uBbRPCtlpTC5s7Q=='".

Worker: [DEBUG    Nuvola] WebExtension.vala:136: Window object cleared for 'https://music.yandex.ru/search?text=the%20xx'
Worker: [DEBUG    Nuvola] WebExtension.vala:144: Init frame: https://music.yandex.ru/search?text=the%20xx, 0x1b9cae0, 0x1a21720, 0x7fb292957140
Runner: Error loading module '/usr/lib/mozilla/plugins/nppdf.so': /usr/lib/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: Error loading module '/usr/lib64/mozilla/plugins/nppdf.so': /usr/lib64/mozilla/plugins/nppdf.so: wrong ELF class: ELFCLASS32
Runner: Error scanning plugin /usr/lib64/mozilla/plugins/nppdf.so, /usr/lib/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-iVHOUO9uBbRPCtlpTC5s7Q==''. It will be ignored.
Runner: https://music.yandex.ru/index.ru.js?v=0.14.02rc01:24:28641: CONSOLE LOG ÐÐ»Ñ Ð¿Ð¾Ð»ÑÑÐµÐ½Ð¸Ñ Ð»Ð¾Ð³Ð° Ð´Ð¸Ð°Ð³Ð½Ð¾ÑÑÐ¸ÐºÐ¸ Ð¾ÑÐ¸Ð±Ð¾Ðº Ð½Ð°Ð±ÐµÑÐ¸ÑÐµ diagnose()
Runner: https://music.yandex.ru/search?text=the%20xx:46: CONSOLE ERROR [Report Only] Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval' vk.com cdn.pushwoosh.com yandex.ru yandex.st yandex.net yastatic.net yandexadexchange.net *.yandex.ru *.yandex.net *.yastatic.net *.yandexadexchange.net *.yandex-team.ru 'nonce-iVHOUO9uBbRPCtlpTC5s7Q=='".

Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

Runner: CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-iVHOUO9uBbRPCtlpTC5s7Q==''. It will be ignored.
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'frame-ancestors'.

Runner: https://music.yandex.ru/index.ru.js?v=0.14.02rc01:25:5016: CONSOLE LOG Ð¡Ð¿ÑÐ°Ð²ÐºÐ° Ð¿Ð¾ Ð¸Ð½ÑÐµÑÑÐµÐ¹ÑÑ Ð´Ð»Ñ ÑÐ°Ð·ÑÐ°Ð±Ð¾ÑÐºÐ¸ ÑÐ°ÑÑÐ¸ÑÐµÐ½Ð¸Ð¹ Ð´Ð¾ÑÑÑÐ¿Ð½Ð° Ð² externalAPI.help();
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-iVHOUO9uBbRPCtlpTC5s7Q==''. It will be ignored.
Vector smash protection is enabled.
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-iVHOUO9uBbRPCtlpTC5s7Q==''. It will be ignored.
Runner: CONSOLE ERROR Unrecognized Content-Security-Policy directive 'manifest-src'.

CONSOLE ERROR The source list for Content Security Policy directive 'script-src' contains an invalid source: ''nonce-iVHOUO9uBbRPCtlpTC5s7Q==''. It will be ignored.
```

**Comment by petRUShka** [comment]
So the really new line I found is:

```
Runner: https://music.yandex.ru/api/v2.1/index/music.yandex.ru:1:2719: CONSOLE ERROR TypeError: undefined is not an object (evaluating 'AudioContext.prototype')
```

It looks like the reason of failure.

**Comment by jiri-janousek** [comment]
Hello @petRUShka. Thanks for your bug report. Could you test with 996782a?

**Comment by petRUShka** [comment]
First of all. It is working with nuvolaplayer3-git (commit `d42792c6d7c83969098da56da9f506f746d05b6f`) and diorite-git (commit `a69a6ef3a0381d945f9cd6bdd1e134ae46599be7`). I misunderstood you and installed git version of player and stable version of yandex-music.

Then I switched back to nuvolaplayer 3.0.4 stable and yandex.music 1.2 stable. That combination fails again.

Then I installed yandex.music-git version and it is working with Yandex.Music from `master`.

**Comment by jiri-janousek** [comment]
Oh, sorry for the confusion, I should've been more clear. Could you test Nuvola Player 3.0.4 with Yandex.Music from master?

**Comment by petRUShka** [comment]
my latest message was confusing :)

Yes, I did and it is working as expected.

**Comment by jiri-janousek** [comment]
Great :-) I'll make a new release then.

**Comment by petRUShka** [comment]
great idea!

**Comment by petRUShka** [comment]
@fenryxo, it would be great if you will release new version. Because it makes possible to upgrade nuvola-yandex-music package for arch linux. Right now it is useless because of white screen.

**Comment by jiri-janousek** [comment]
I might get down to it on Wednesday.

**Comment by jiri-janousek** [comment]
Fixed in 1.3.
